qa/rpc-tests/wallet-dump.py

```diff
@@ -81,12 +92,19 @@ def run_test (self):
         self.nodes[0].keypoolrefill()
 
         # dump unencrypted wallet
-        self.nodes[0].dumpwallet(tmpdir + "/node0/wallet.unencrypted.dump")
+        key = None
+        try:
+            self.nodes[0].dumpwallet(tmpdir + "/node0/wallet.unencrypted.dump")
+        except Exception as ex:
+            key = get_dumpwallet_otp (ex.error['message'])
+            self.nodes[0].dumpwallet(tmpdir + "/node0/wallet.unencrypted.dump", key)
+        assert key, 'Import wallet did not raise exception when was called first time without one-time code.'
 
         found_addr, found_addr_chg, found_addr_rsv, hd_master_addr_unenc = \
             read_dump(tmpdir + "/node0/wallet.unencrypted.dump", addrs, None)
         assert_equal(found_addr, test_addr_count)  # all keys must be in the dump
-        assert_equal(found_addr_chg, 50)  # 50 blocks where mined
+
+        assert_equal(found_addr_chg, 50 + hdmint_key_count) # 50 block were mined + hdmint keys
```

Comment:
@riordant do you know why it treat HD mint key as a change key? This does not seem right to me. For me it should be `reserve=1` instead of `change=1` for HD mint key in the pool.